buffer_pool.go

```diff
@@ -41,7 +41,9 @@ func (b *bufferPool) Get() *bytes.Buffer {
 }
 
 func (b *bufferPool) Put(buf *bytes.Buffer) {
-	if buf.Cap() > maxRecycleBufferSize {
+	// Don't hold onto buffers that are too large.
+	// Or buffers that have no capacity.
+	if buf.Cap() == 0 || buf.Cap() > maxRecycleBufferSize {
```

Comment:
For buffer pools ive been keeping the topmost defer and then when we need to extend it replace the buffer with an empty one: `*buf = bytes.Buffer{}`. Then the `pool.Put()` is a no-op and we can create a new buffer when we need to return it.